jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/S3ObjectAction.kt

```diff
@@ -23,7 +23,17 @@ abstract class S3ObjectAction(protected val treeTable: S3TreeTable, title: Strin
 
     override fun actionPerformed(e: AnActionEvent) = performAction(selected().filter { it !is S3TreeContinuationNode })
 
-    private fun selected(): List<S3TreeNode> = treeTable.getSelectedNodes().takeIf { it.isNotEmpty() } ?: listOf(treeTable.getRootNode())
+    private fun selected(): List<S3TreeNode> {
+        val selectedNodes = treeTable.getSelectedNodes()
+        if (selectedNodes.isNotEmpty()) {
+            return selectedNodes
+        }
+        val rootNode = treeTable.getRootNode()
+        if (rootNode != null) {
```

Comment:
`listOfNotNull(rootNode)`

Reply:
Now that rootNode is not nullable again this change can be reverted